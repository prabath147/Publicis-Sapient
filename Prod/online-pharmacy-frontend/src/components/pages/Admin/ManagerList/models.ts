import { PageableResponse } from "../storeList/models";

export interface Manager 
{
  managerId: number;
  name: string;
  phoneNo: string;
  licenseNo: string;
  registrationDate: Date;
  approvalStatus: "APPROVED" | "PENDING" | "REJECTED";
}


export interface ManagerListResponse extends PageableResponse{
  data:Manager[]
}