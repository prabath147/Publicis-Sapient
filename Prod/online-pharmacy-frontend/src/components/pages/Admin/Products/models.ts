
export interface CategorySet {
  categoryId: number;
  categoryName: string;
}
export interface Product {
  categorySet: CategorySet[];
  description: string;
  dosageForm: string;
  imageUrl: string;
  productId: number;
  productName: string;
  productType: boolean;
  proprietaryName: string;
  quantity: number;
}

