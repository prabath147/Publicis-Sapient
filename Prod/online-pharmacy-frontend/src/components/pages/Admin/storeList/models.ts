import { Item } from "../../Manager/models";

export interface Test {
  data: Store[];
  pageNumber: number;
  pageSize: number;
  totalRecords: number;
  totalPages: number;
  isLastPage: boolean;
}


export interface Store {
  storeId: number;
  storeName: string;
  address: Address;
  createdDate: string;
  manager: Manager;
  revenue: number;
}

export interface Address {
  addressId: number;
  street: string;
  city: string;
  state: string;
  pinCode: number;
  country: string;
}

export interface Manager {
  managerId: number;
  name: string;
  phoneNo: string;
  licenseNo: string;
  registrationDate: string;
  approvalStatus: string;
}

export interface PageableResponse{
  "pageNumber":number,
  "pageSize":number,
  "totalRecords":number,
  "totalPages":number,
  "isLastPage":boolean
}

export interface getStoreInventoryResponse extends PageableResponse{
  data : Item[]
}


