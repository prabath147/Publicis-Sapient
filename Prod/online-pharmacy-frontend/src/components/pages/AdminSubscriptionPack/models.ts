export interface Subscriptionv{
    subscriptionId:number,
    subscriptionName:string,
    description:string,
    startDate:Date,
    endDate:Date,
    benefits:Benefit
    subscriptionCost:number,
    period:number,
    status:string
}

export interface Benefit{
    delivery_discount:number,
    one_day_delivery:boolean
}
export interface SubDetails{
    subscriptionName:string,
    description:string,
    startDate:Date,
    endDate:Date,
    benefits:Benefit,
    subscriptionCost:number,
    period:number
}

export interface SubRequest{
    name:string,
    description:string,
    stDate:Date,
    edDate:Date,
    deliveryDiscount:number,
    oneDayDelivery:string,
    cost:number,
    period:number
}