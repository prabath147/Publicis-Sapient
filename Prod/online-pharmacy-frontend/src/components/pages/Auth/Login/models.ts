export interface User {
  id: number;
  username: string;
  email: string;
  role: string;
}

export interface LoginRequest {
  username: string;
  password: string;
}

export interface LoginResponse {
  id: number;
  username: string;
  email: string;

  token: string;
  refreshToken: string;
}

export interface tokenStructure {
  sub: string;
  role: string;
  iat: number;
  exp: number;
  id:number;
}
