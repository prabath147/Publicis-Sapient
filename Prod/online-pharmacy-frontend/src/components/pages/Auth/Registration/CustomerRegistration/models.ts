import { AddressInterface } from './../../../../ui/forms/AddressForm';

export interface CustomerRegistrationRequest {
  username: string;
  email: string;
  password: string;
  role: ["customer"];

  detailObject: {
    fullName: string;
    mobileNumber: number;
    address: AddressInterface
  };
}

export interface CustomerRegistrationForm {
  fullName: string;
  username: string;
  mobileNumber: number;
  email: string;
  password: string;
  confirmPassword: string;
  address: AddressInterface
}
