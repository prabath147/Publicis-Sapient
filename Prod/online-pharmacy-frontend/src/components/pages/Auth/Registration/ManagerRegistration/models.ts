export interface ManagerRegistrationRequest {
  username: string;
  email: string;
  password: string;
  role: ["manager"];

  detailObject: {
    name: string;
    phoneNo: number;
    licenseNo: number;
  };
}

export interface ManagerRegistrationForm {
  username: string;
  email: string;
  password: string;
  confirmPassword: string;
  name: string;
  phoneNo: number;
  licenseNo: number;
}
