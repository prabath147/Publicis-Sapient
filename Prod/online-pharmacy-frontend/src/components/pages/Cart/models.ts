import { AddressInterface } from "../../ui/forms/AddressForm";

export interface ItemMiniDTO {
  itemId?: number;
  itemIdFk: number;
  itemQuantity: number;
  price: number;
}

export interface ProductMiniDTO {
  productId?: number;
  productIdFk: number;
  quantity: number;
  price: number;
}

export interface CartSliceType {
  items: ItemMiniDTO[];
  selected: number[],
  totalAmount: number,
  totalItem: number,
}


export interface ItemDetailDTO {
  expiryDate: Date;
  itemId: number;
  itemQuantity: number;
  manufacturedDate: string;
  price: number;
  product: ProductDTO;
  store: Store;
}

export interface ProductDTO {
  categorySet: CategorySet[];
  description: string;
  dosageForm: string;
  imageUrl: string;
  productId: number;
  productName: string;
  productType: boolean;
  proprietaryName: string;
  quantity: number;
}

export interface CategorySet {
  categoryId: number;
  categoryName: string;
}

export interface Store {
  address: AddressInterface;
  createdDate: string;
  manager: Manager;
  revenue: number;
  storeId: number;
  storeName: string;
}

export interface Manager {
  approvalStatus: string;
  licenseNo: string;
  managerId: number;
  name: string;
  phoneNo: string;
  registrationDate: string;
}

export interface ProductDetailDTO {
  description: string;
  dosageForm: string;
  id: string;
  imageUrl: string;
  itemId: number;
  price: number;
  productId: number;
  productName: string;
  productType: boolean;
  proprietaryName: string;
  quantity: number;
  storeId: number;
}

// Generated by https://quicktype.io


export interface PrescriptionItem {
  id: number;
  name: string;
  quantity: number;
  price: number;
}