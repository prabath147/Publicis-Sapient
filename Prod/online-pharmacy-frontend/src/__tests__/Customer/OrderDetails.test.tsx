import { render, waitForElementToBeRemoved } from "@testing-library/react"
import OrderDetailPage from "../../components/pages/orderDetail/OrderDetailPage"
import { Order } from "../../components/pages/orderList/models"
import { addressInitialValues } from "../../components/ui/forms/AddressForm"
import mockAxios from "../../__mocks__/axios"

const simpleCard = ({ item, children, checkbox }) => {
    return <>
        {children}
        {checkbox}
    </>
}

jest.mock("../../components/pages/item/CartItemCard", () => () => (<>Component</>))

const MockOrderDetails = () => {
    return <OrderDetailPage />
}

const orderData: Order = {
    items: [{
        itemIdFk: -1,
        itemQuantity: 1,
        price: 1,
        itemId: 1
    }],
    optionalOrderDetails: false,
    orderAddress: addressInitialValues(),
    orderDate: new Date('2020-01-01'),
    orderId: 1,
    price: 1,
    quantity: 1,
    userId: 1,
    orderDetails: {
        orderDetailsId: 1
    }
}

describe("OrderDetailPage", () => {
    afterEach(() => {
        jest.resetAllMocks();
    });
    it("should create a new OrderDetailPage", () => {
        mockAxios.get.mockImplementationOnce(() => Promise.reject());
        const { asFragment } = render(
            <MockOrderDetails />
        )
        expect(asFragment()).toMatchSnapshot()
    })

    it("should create a new on fail", async () => {
        mockAxios.get.mockImplementationOnce(() => Promise.resolve({
            data: orderData
        }))


        const { asFragment, getByTestId } = render(
            <MockOrderDetails />
        )

        await waitForElementToBeRemoved(getByTestId(/Loading/i));

        expect(asFragment()).toMatchSnapshot()
    })
})