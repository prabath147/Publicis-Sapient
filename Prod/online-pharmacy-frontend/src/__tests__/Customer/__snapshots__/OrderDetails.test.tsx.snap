// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetailPage should create a new OrderDetailPage 1`] = `
<DocumentFragment>
  <div
    class="mantine-8od8ev"
    style="height: 300px;"
  >
    <svg
      class="mantine-1avyp1d"
      data-testid="Loading"
      height="36px"
      role="presentation"
      stroke="#228be6"
      viewBox="0 0 38 38"
      width="36px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <g
          stroke-width="5"
          transform="translate(2.5 2.5)"
        >
          <circle
            cx="16"
            cy="16"
            r="16"
            stroke-opacity=".5"
          />
          <path
            d="M32 16c0-9.94-8.06-16-16-16"
          >
            <animatetransform
              attributeName="transform"
              dur="1s"
              from="0 16 16"
              repeatCount="indefinite"
              to="360 16 16"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>
</DocumentFragment>
`;

exports[`OrderDetailPage should create a new on fail 1`] = `
<DocumentFragment>
  <div>
    <div
      class="mantine-Container-root mantine-15pjuqq"
    >
      <h1
        class="mantine-Text-root mantine-Title-root mantine-dyfpg1"
      >
        Order Details
      </h1>
      <div
        class="mantine-Group-root mantine-yrdz4u"
      >
        <div
          class="mantine-Text-root mantine-Group-child mantine-1wsjsij"
        >
          Ordered on Wed Jan 01 2020 05:30:00 GMT+0530 (India Standard Time)
        </div>
        <div
          class="mantine-Text-root mantine-Group-child mantine-1wsjsij"
        >
          Order# 1
        </div>
      </div>
      <div
        class="mantine-Divider-root mantine-Divider-horizontal mantine-polw48"
        role="separator"
      />
      <div
        class="mantine-Paper-root mantine-Card-root mantine-xoj88x"
      >
        <div
          class="mantine-Grid-root mantine-1xetgn3"
        >
          <div
            class="mantine-Col-root mantine-1r1hpk"
          >
            <h5
              class="mantine-Text-root mantine-Title-root mantine-1cdc0tb"
            >
              Shipping Address
            </h5>
            <div
              class="mantine-Text-root mantine-1pbxw0k"
            >
              , , ,
              <br />
              , 
            </div>
          </div>
          <div
            class="mantine-Col-root mantine-1r1hpk"
          >
            <h5
              class="mantine-Text-root mantine-Title-root mantine-1cdc0tb"
            >
              Payment Method
            </h5>
            <div
              class="mantine-Text-root mantine-1pbxw0k"
            >
              Pay on Delivery (Cash/Card). Cash on delivery (COD) available. Card/Net banking acceptance subject to device availability.
            </div>
          </div>
          <div
            class="mantine-Col-root mantine-1r1hpk"
          >
            <h5
              class="mantine-Text-root mantine-Title-root mantine-1cdc0tb"
            >
              Order Summary
            </h5>
            <div
              class="mantine-Group-root mantine-aa1sbi"
            >
              <div
                class="mantine-Text-root mantine-Group-child mantine-1wsjsij"
              >
                Item(s) Subtotal:
              </div>
              <div
                class="mantine-Text-root mantine-Group-child mantine-1wsjsij"
              >
                 ₹1
              </div>
            </div>
            <div
              class="mantine-Group-root mantine-aa1sbi"
            >
              <div
                class="mantine-Text-root mantine-Group-child mantine-1wsjsij"
              >
                Shipping:
              </div>
              <div
                class="mantine-Text-root mantine-Group-child mantine-1wsjsij"
              >
                 ₹ 100
              </div>
            </div>
            <div
              class="mantine-Group-root mantine-aa1sbi"
            >
              <div
                class="mantine-Text-root mantine-Group-child mantine-1wsjsij"
              >
                Promotion Applied:
              </div>
              <div
                class="mantine-Text-root mantine-Group-child mantine-1wsjsij"
              >
                 -100 
              </div>
            </div>
            <div
              class="mantine-Group-root mantine-aa1sbi"
            >
              <div
                class="mantine-Text-root mantine-Group-child mantine-738zg2"
              >
                Grand Total:
              </div>
              <div
                class="mantine-Text-root mantine-Group-child mantine-738zg2"
              >
                 1 
              </div>
            </div>
          </div>
        </div>
      </div>
      Component
    </div>
  </div>
</DocumentFragment>
`;
